jetty-core/jetty-websocket/jetty-websocket-jetty-client/src/main/java/org/eclipse/jetty/websocket/client/internal/DelegatedJettyClientUpgradeRequest.java

```diff
@@ -78,7 +79,7 @@ public List<String> getHeaders(String name)
     @Override
     public Map<String, List<String>> getHeaders()
     {
-        return null;
+        return Collections.unmodifiableMap(HttpFields.asMap(delegate.getHeaders()));
```

Comment:
Does it need to be unmodifieable?
Are we sure the delegate is still valid when this is called? Perhaps do in the constructor?

Reply:
Now the `asMap` method now determines whether it is immutable or not based on whether an instance of `HttpFields` was provided or `HttpFields.Mutable`